kyo-kernel/jvm/src/main/scala/kyo/kernel/internal/TracePool.scala

```diff
@@ -17,28 +18,33 @@ private[kernel] object TracePool:
 
         final def borrow(): Trace =
             if size == 0 then
-                if global.drain(add, localCapacity - size) == 0 then
-                    return Trace.init
-            size -= 1
-            val buffer = pool(size)
-            pool(size) = null
-            buffer
+                discard(global.drain(add, localCapacity))
```

Comment:
I've also made minor optimizations to avoid reading the `size` field multiple times